chia/wallet/dao_wallet/dao_wallet.py

```diff
@@ -0,0 +1,2268 @@
+                        if cat_coin == cat_coins[-1]:  # the last coin is the one that makes the conditions
+                            if sum_of_coins - sum_of_conditions > 0:
+                                change_condition = Program.to(
+                                    [
+                                        51,
+                                        p2_singleton_puzzle.get_tree_hash(),
+                                        sum_of_coins - sum_of_conditions,
+                                        [self.dao_info.treasury_id],
+                                    ]
+                                )
+                                delegated_puzzle = Program.to((1, change_condition.cons(conditions)))
+                            else:  # pragma: no cover
+                                delegated_puzzle = Program.to((1, conditions))
+
+                            solution = Program.to(
+                                [
+                                    0,
+                                    treasury_inner_puzhash,
+                                    delegated_puzzle,
+                                    0,
+                                    cat_coin.name(),
+                                ]
+                            )
+                        else:
+                            solution = Program.to(
+                                [
+                                    0,
+                                    treasury_inner_puzhash,
+                                    0,
+                                    0,
+                                    cat_coin.name(),
+                                ]
+                            )
+                        new_spendable_cat = SpendableCAT(
+                            cat_coin,
+                            tail_hash,
+                            p2_singleton_puzzle,
+                            solution,
+                            lineage_proof=lineage_proof,
+                        )
+                        spendable_cat_list.append(new_spendable_cat)
+                    # create or merge with other CAT spends
+                    if cat_spend_bundle is None:
+                        cat_spend_bundle = unsigned_spend_bundle_for_spendable_cats(CAT_MOD, spendable_cat_list)
+                    else:
+                        cat_spend_bundle = cat_spend_bundle.aggregate(
+                            [cat_spend_bundle, unsigned_spend_bundle_for_spendable_cats(CAT_MOD, spendable_cat_list)]
+                        )
+                    tailhash_parent_amount_list.append([tail_hash, parent_amount_list])
+
+                delegated_solution = Program.to(
+                    [
+                        xch_parent_amount_list,
+                        tailhash_parent_amount_list,
+                        treasury_inner_puzhash,
+                    ]
+                )
+
+            elif proposal_type == ProposalType.UPDATE:
+                (
+                    _,
```

Comment:
These underscores come out of nowhere and I can't find where they're defined.  It'd be good to give this constant value a name.

Reply:
fixed